sorting/pigeonhole_sort.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Implementation of [Pigeonhole Sort algorithm]
+ * (https://en.wikipedia.org/wiki/Pigeonhole_sort)
+ * @author [Lownish](https://github.com/Lownish)
+ * @details
+ * Pigeonhole sorting is a sorting algorithm that is suitable for sorting lists
+ * of elements where the number of elements and the number of possible key
+ * values are approximately the same. It requires O(n + Range) time where n is
+ * number of elements in input array and ‘Range’ is number of possible values in
+ * array.
+ *
+ * The time Complexity of the algorithm is \f$O(n+N)\f$.
+ */
+
+#include <algorithm>  //for std::is_sorted
+#include <array>      //for std::array
+#include <cassert>    //for assert
+#include <iostream>   //for io operations
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace pigonhole
+ * @brief Functions for [Pigeonhole
+ * Sort](https://en.wikipedia.org/wiki/Pigeonhole_sort) algorithm
+ */
+namespace pigonhole {
+/**
+ * Pigeonhole sorting of array of size n
+ * The function will sort the array through Pigeonhole algorithm and print
+ * @param arr unsorted array of elements
+ * @returns none
+ */
+template <std::size_t N>
+void pigeonSort(std::array<int, N> arr, int n) {
+    // Finding min and max*
+    int min = arr.at(0);
+    int max = arr.at(0);
+
+    for (int i = 0; i < n; i++) {
+        if (min > arr[i]) {
+            min = arr[i];
+        }
+        if (max < arr[i]) {
+            max = arr[i];
+        }
+    }
+
+    // Range refers to the number of holes required
+    int range = max - min + 1;
+    int *hole = new int[range]();
+
+    // Copying all array values to pigeonhole
+    for (int i = 0; i < n; i++) {
+        hole[arr[i] - min] = arr[i];
+    }
+
+    // Deleting elements from list and storing to original array
+    int count = 0;
+    for (int i = 0; i < range; i++) {
+        while (hole[i] != '\0') {
+            arr[count] = hole[i];
+            hole[i] = {};
+            count++;
+        }
+    }
+    delete[] hole;
+    assert(std::is_sorted(std::begin(arr), std::end(arr)));
+    std::cout << "Passed\n";
+
+    // Printing sorted array
+    for (int i = 0; i < n; i++) {
+        std::cout << arr.at(i) << " ";
+    }
+}
+}  // namespace pigonhole
+}  // namespace sorting
+
+/**
+ * Test function 1 with unsorted array
+ * {8, 3, 2, 7, 4, 6, 8}
+ * @returns none
+ */
+static void test_1() {
+    const int n = 7;
+    std::array<int, n> test_array = {8, 3, 2, 7, 4, 6, 8};
+
+    sorting::pigonhole::pigeonSort(test_array, n);
```

Comment:
```suggestion
    sorting::pigeonhole::pigeonSort<n>(test_array, n);
```

Reply:
The "<n>" part is giving errors. Without this part, it is working fine

Reply:
I have tested it and compiled it with CMake and it has no errors.
Please specify the error that the linter is throwing.